packages/wallet-frontend/src/utils/debounce.ts

```diff
@@ -0,0 +1,16 @@
+import { useEffect, useState } from 'react'
+
+export default function useDebounce(value: string, delay: number) {
+  const [debouncedValue, setDebouncedValue] = useState(value)
+
+  useEffect(() => {
+    const handler = setTimeout(() => {
+      setDebouncedValue(value)
+    }, delay)
+
+    return () => {
+      clearTimeout(handler)
+    }
+  }, [value, delay])
+  return debouncedValue
```

Comment:
What do you think about storing the previous value? In our case, if a user pastes the same URL into the input it would not make another call to the backend.

Reply:
ALso a great idea, will try to implement, but the previous value should be verified in the pay.tsx and not in the debounce, no?

Reply:
It would make sense to be in the `useDebounce` hook. Thinking more about this, we can also return a `isLoading` property to display a loading state while we are waiting for the debounced value.